agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4251,7 +4254,67 @@ bool slave_thread::handle_client_capability_query(Socket *sd, ieee1905_1::CmduMe
 {
     const auto mid = cmdu_rx.getMessageId();
     LOG(DEBUG) << "Received CLIENT_CAPABILITY_QUERY_MESSAGE , mid=" << std::dec << int(mid);
-    return true;
+
+    auto client_info_tlv_r = cmdu_rx.getClass<wfa_map::tlvClientInfo>();
+    if (!client_info_tlv_r) {
+        LOG(ERROR) << "getClass wfa_map::tlvClientInfo failed";
+        return false;
+    }
+
+    //Check if it is an error scenario - if the STA specified in the Client Capability Query message is not associated
+    //with any of the BSS operated by the Multi-AP Agent (an error scenario)
+    bool error_scenario = 1;
+    auto it             = m_vaps_clients_map.find(client_info_tlv_r->bssid());
+    if (it != m_vaps_clients_map.end()) {
+        if (it->second.count(client_info_tlv_r->client_mac())) {
+            //only if the client exist in one of the BSS operated by the Multi-AP Agent this is not an error scenario
+            error_scenario = 0;
+        }
+    }
```

Comment:
This is not accurate... If the client is associated with a BSS on another radio in the same agent, both radios will receive the query, but it is not associated with this radio. We shouldn't treat that as an error.

Also, the bssid given in the client info should be treated as just a hint. The spec says "If the STA specified in the Client Capability Query message is not associated with any of the BSS operated by the Multi-AP Agent". It doesn't say "if it is not associated with the BSS given by the client info TLV". A likely scenario is that a client associates and immediately disassociates and moves to another band, so it's possible that the controller sends the query still using the old BSS (which it got from the topology notification).

Bottom line, I think the best approach is to again cache everything on the backhaul_manager and generate the message there.